netmaster/master/netmaster.go

```diff
@@ -501,17 +501,26 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 		// XXX: do configuration check, to make sure it is allowed
 	}
 
+	subnetIsAllocated := false
 	if nwCfg.SubnetIP == "" {
 		nwCfg.SubnetLen = gCfg.Auto.AllocSubnetLen
 		nwCfg.SubnetIP, err = gCfg.AllocSubnet(rm)
 		if err != nil {
 			return err
 		}
+		subnetIsAllocated = true
 	}
 
 	nwCfg.DefaultGw = network.DefaultGw
-	if nwCfg.DefaultGw == "" {
-		// TBD: allocate per global policy
+	// For auto derived subnets assign gateway ip be the last valid unicast ip the subnet
+	if nwCfg.DefaultGw == "" && subnetIsAllocated {
```

Comment:
> What we need is a mechanism that can allow specifying which one is the default interface among multiple. Can that be done as a separate PR?

Sure, this change, though small, adds a feature imo. I am not sure if we should add just a part of feature, but up to you.

> About unit tests:

@jainvipin I actually meant adding a few system-tests. I was thinking we should be able to test this feature may be by a ping to default-gateway or routing between networks?